external/storm-solr/src/main/java/org/apache/storm/solr/bolt/SolrUpdateBolt.java

```diff
@@ -92,9 +108,17 @@ private void ack(Tuple tuple) throws SolrServerException, IOException {
         if (commitStgy == null) {
             collector.ack(tuple);
         } else {
-            toCommitTuples.add(tuple);
-            commitStgy.update();
-            if (commitStgy.commit()) {
+            boolean forceCommit = false;
+            if (TupleUtils.isTick(tuple)) {
+                LOG.debug("TICK! forcing solr client commit");
+                collector.ack(tuple);
+                forceCommit = true;
+            } else {
+                toCommitTuples.add(tuple);
+                commitStgy.update();
+            }
+
+            if (forceCommit || commitStgy.commit()) {
```

Comment:
@vesense Can you please follow te suggestion that I gave as the code is much cleaner.